nbconvert/exporters/pdf.py

```diff
@@ -175,7 +175,7 @@ def from_notebook_node(self, nb, resources=None, **kw):
                 rc = self.run_latex(tex_file)
             
             pdf_file = notebook_name + '.pdf'
-            if not os.path.isfile(pdf_file):
+            if not rc or not os.path.isfile(pdf_file):
```

Comment:
I think I see the issue. Look at the code a few lines above. The first ```rc = self.run_latex(tex_file)``` succeeds. Then:
```python
if rc:
     rc = self.run_bib(tex_file)
```
runs and fails which with your change causes tests to think they are failing.

I think saving the first return code as ```latex_rc = rc = self.run_latex(tex_file)``` and then doing
```python
if not latex_rc or …
```
should work.

Reply:
@t-makaro I think optimally `self.run_bib(tex_file)` shouldn't be run if there is no bibliographic data, but I agree that your approach works in those cases. As I understand it, wouldn't it ignore legitimate bibtex failures too?

Reply:
Yes, the bibtex should not be run if there is no bibliographic data. I'm not sure how the bibtex integration works. Detecting the existence of a bibtex file may not be enough.

I think that just checking for latex failures would catch/display >95% of errors, and it would not harm any future endeavour to catch legitimate bibtex errors which are not caught now anyway. IMHO, catching only latex errors should be merged for the 5.5 release as it would solve a very long standing issue.

**CC** @MSeal

Reply:
I think it's fair to solve for latex issues first and tackle bibtex later. I too am not familiar with the bibtex integration enough to say how it should work here without some research into the problem.

The plan is to get 5.5 out here in April. If you want to get tests passing and then try tackling the biliographic failure catches in a follow-up thread we can get at least the minimal improvement. If you think you see a clear path to catching both that your confident in I'm ok with that too. Up to you on what you can fix before we release in a week or so. Tests will need to be passing for a merge either way.

Reply:
Also thank you for digging in to find the underlying issue. Been needing some eyes on these issues for a bit now.